src/test/java/space/obminyashka/items_exchange/service/UserServiceTest.java

```diff
@@ -45,45 +47,70 @@ class UserServiceTest {
     private RoleService roleService;
     @Mock
     private UserMapper userMapper;
-    @Captor
-    private ArgumentCaptor<String> oauth2UserArgumentCaptor;
     private UserServiceImpl userService;
     @Value("${number.of.hours.to.keep.email.confirmation.code}")
     private int numberOfHoursToKeepEmailConformationToken;
 
     @BeforeEach
     void setUp() {
-        userService = new UserServiceImpl(bCryptPasswordEncoder,userRepository, emailConfirmationCodeRepository, phoneMapper, roleService, userMapper, numberOfHoursToKeepEmailConformationToken);
+        userService = new UserServiceImpl(bCryptPasswordEncoder, userRepository, emailConfirmationCodeRepository, phoneMapper, roleService, userMapper, numberOfHoursToKeepEmailConformationToken);
     }
 
     @Test
-    void testLoginUserWithOAuth2_WhenUserBeenCreated() {
-        var oauth2User = createDefaultOidcUser();
-        when(userRepository.findByEmailOrUsername(NEW_USER_EMAIL, NEW_USER_EMAIL)).thenReturn(creatOptionalUser());
-        userService.loginUserWithOAuth2(oauth2User);
-
-        verify(userRepository).setOAuth2LoginToUserByEmail(oauth2UserArgumentCaptor.capture());
+    void testLoginUserWithOAuth2_WhenUserBeenCreatedWithFalseValidatedEmail() {
+        //Arrange
 
-        assertEquals(NEW_USER_EMAIL, oauth2UserArgumentCaptor.getValue());
+        var oauth2User = createDefaultOidcUser();
+        var optionalUserProjection = creatOptionalUserProjection();
+        when(userProjection.getEmail()).thenReturn(NEW_USER_EMAIL);
+        when(userProjection.getOauth2Login()).thenReturn(false);
+        when(userProjection.getIsValidatedEmail()).thenReturn(false);
+        when(userRepository.findUserProjectionByEmail(NEW_USER_EMAIL)).thenReturn(optionalUserProjection);
```

Comment:
Why don't you populate 'userProjection' with the data and simply return it via mocked repository instead of mocking every field?